src/Repeater/RepeaterServer.ts

```diff
@@ -0,0 +1,72 @@
+import { Protocol } from '../RequestExecutor';
+
+export interface RepeaterServerDeployedEvent {
+  repeaterId: string;
+}
+
+export interface RepeaterServerRequestEvent {
+  protocol: Protocol;
+  url: string;
+  method?: string;
+  headers?: Record<string, string | string[]>;
+  correlationIdRegex?: string;
+  body?: string;
+}
+
+export interface RepeaterServerRequestResponse {
+  protocol: Protocol;
+  statusCode?: number;
+  message?: string;
+  errorCode?: string;
+  headers?: Record<string, string | string[] | undefined>;
+  body?: string;
+}
```

Comment:
Let's use the union type to incorporate better type-safety

```suggestion
export interface RepeaterServerRequestResponse =
  | {
      protocol: Protocol;
      statusCode?: number;
      message?: string;
      errorCode?: string;
      headers?: Record<string, string | string[] | undefined>;
      body?: string;
    }
  | {
      protocol: Protocol;
      message?: string;
      errorCode?: string;
    };
```